sell_only_by_packaging/models/product_template.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    sell_only_by_packaging = fields.Boolean(
+        string="Only sell by packaging",
+        company_dependent=True,
+        default=False,
+        help="Restrict the usage of this product on sale order lines without "
+        "packaging defined",
+    )
+
+    min_sellable_qty = fields.Float(
+        compute="_compute_template_min_sellable_qty",
+        help=(
+            "Minimum sellable quantity, according to the available packagings, "
+            "if Only Sell by Packaging is set."
+        ),
+    )
+
+    @api.depends(
+        "sell_only_by_packaging",
+        "uom_id.factor",
+        "product_variant_ids.min_sellable_qty",
+    )
+    def _compute_template_min_sellable_qty(self):
+        for record in self:
+            record.min_sellable_qty = 0.0
+            if len(record.product_variant_ids) == 1:
+                # Pick the value from the variant if there's only 1
+                record.min_sellable_qty = record.product_variant_ids.min_sellable_qty
+
+    @api.constrains("sell_only_by_packaging", "sale_ok")
+    def _check_sell_only_by_packaging_sale_ok(self):
+        for product in self:
+            if product.sell_only_by_packaging and not product.sale_ok:
+                raise ValidationError(
+                    _(
```

Comment:
```suggestion
                    self.env._(
```
same with all `_(` above code